### ISSUE TITLE
UX Improvement: Enhance Accessibility and Interaction in the Terms of Use Dialog

### ISSUE THREAD

**Issue by annavilnid** [compolabs/spark-frontend#388]
<img width="555" alt="Image" src="https://github.com/user-attachments/assets/77852939-c8ce-4ef5-8a21-c1d65eed16f5" />

1️⃣ The "AGREE AND CONTINUE" button remains active even if the "I accept the Terms of Use" checkbox is unchecked.

2️⃣ The text is structured using  div instead of semantic tags for text, which reduces accessibility and SEO.

3️⃣ The close button (❌) is implemented as an svg instead of a <button>, making it inaccessible via keyboard navigation (Tab, Enter).

4️⃣ The "I acknowledge that I am trading a real money" checkbox cannot be disabled.

5️⃣ The checkboxes cannot be navigated using the keyboard (Tab, Enter).

6️⃣ Alt text is not used for icons and images, which reduces accessibility. For example, the close button (❌) could have aria-label="Close" and an alternative text inside the svg.
